adding haven60 pcb firmware

CMMS-Freather opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->
adding haven60 pcb firmware
<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
adding haven60 pcb firmware
<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

waffle87 commented on keyboards/ah/haven60/config.h:
This file should be removed.

waffle87 commented on keyboards/ah/haven60/haven60.c:
```diff
@@ -0,0 +1,42 @@
+);
+
+
+void keyboard_post_init_user(void){
+    rgblight_layers = my_rgb_layers;
+}
+
+bool led_update_user(led_t led_state){
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, !led_state.caps_lock);
+    return true;
+}
```
```suggestion
void keyboard_post_init_kb(void){
    rgblight_layers = my_rgb_layers;
    keyboard_post_init_user();
}

bool led_update_kb(led_t led_state){
    bool res = led_update_user(state);
    if (res) {
        rgblight_set_layer_state(0, led_state.caps_lock);
        rgblight_set_layer_state(1, !led_state.caps_lock);
    }
    return res;
}
```

waffle87 commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,114 @@
+        "led_count": 13,
+        "saturation_steps": 8,
+        "brightness_steps": 8,
+        "sleep": true,
```
```suggestion
        "sleep": true,
        "max_brightness": 125,
        "layers": {
            "enabled": true
        },
```

waffle87 commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,114 @@
+            "static_gradient": true,
+            "twinkle": true
+        }
+    },
```
```suggestion
    },
    "community_layouts": ["60_ansi"],
```

zvecr replied:
```suggestion
    },
    "community_layouts": ["60_ansi_tsangan"],
```

waffle87 commented on keyboards/ah/haven60/keymaps/via/rules.mk:
```diff
@@ -0,0 +1,2 @@
+# This file intentionally left blank
```
Not blank is it?
```suggestion
```

waffle87 commented on keyboards/ah/haven60/readme.md:
```diff
@@ -0,0 +1,27 @@
+# haven60
+
+![haven60](https://i.imgur.com/Uja750O.png)
```
```suggestion
![haven60](https://i.imgur.com/Uja750Oh.png)
```

waffle87 commented on keyboards/ah/haven60/readme.md:
```diff
@@ -0,0 +1,27 @@
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [Freather](https://github.com/Freather)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
These fields must be filled out.

waffle87 commented on keyboards/ah/haven60/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+Make example for this keyboard (after setting up your build environment):
+
+    make haven60:default
+
+Flashing example for this keyboard:
+
+    make haven60:default:flash
```
```suggestion
    make ah/haven60:default

Flashing example for this keyboard:

    make ah/haven60:default:flash
```

fauxpark commented on keyboards/ah/haven60/haven60.c:
```diff
@@ -0,0 +1,46 @@
+}
+
+bool led_update_kb(led_t led_state){
+    bool res = led_update_user(state);
```
```suggestion
    bool res = led_update_user(led_state);
```

zvecr commented on keyboards/ah/haven60/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+![haven60](https://i.imgur.com/Uja750Oh.png)
+
+*A short description of the keyboard/project*
```
This needs to be filled in

CMMS-Freather replied:
updated

zvecr replied:
Not sure why we would still want the template text...
```suggestion
```

zvecr commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,118 @@
+                {"matrix": [3, 11], "x": 11.25, "y": 3},
+                {"matrix": [3, 12], "x": 12.25, "y": 3, "w": 2.75},
+                {"matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
+                {"matrix": [4, 1], "x": 1.55, "y": 4, "w": 1},
```
```suggestion
                {"matrix": [4, 1], "x": 1.5, "y": 4, "w": 1},
```

zvecr commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,118 @@
+                {"matrix": [4, 6], "x": 4, "y": 4, "w": 7},
+                {"matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
+                {"matrix": [4, 12], "x": 12.5, "y": 4, "w": 1},
+                {"matrix": [4, 13], "x": 14, "y": 4, "w": 1.5}
```
```suggestion
                {"matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
```

zvecr commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,118 @@
+        }
+    },
+    "layouts": {
+        "LAYOUT": {
```
```suggestion
        "LAYOUT_60_ansi_tsangan": {
```

zvecr commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,34 @@
+     * │Ctrl│GUI │Alt │                        │ Alt│   GUI   │Ctrl│
+     * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+     */
+    [0] = LAYOUT(
```
```suggestion
    [0] = LAYOUT_60_ansi_tsangan(
```

zvecr commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,34 @@
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, 
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                     KC_RALT, MO(1),  KC_RCTL 
+    ),
+    [1] = LAYOUT(
```
```suggestion
    [1] = LAYOUT_60_ansi_tsangan(
```

zvecr commented on keyboards/ah/haven60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,34 @@
+     * │Ctrl│GUI │Alt │                        │ Alt│   GUI   │Ctrl│
+     * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+     */
+    [0] = LAYOUT(
```
```suggestion
    [0] = LAYOUT_60_ansi_tsangan(
```

zvecr commented on keyboards/ah/haven60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,34 @@
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                     KC_RALT, MO(1),  KC_RCTL
+    ),
+    [1] = LAYOUT(
```
```suggestion
    [1] = LAYOUT_60_ansi_tsangan(
```

zvecr commented on keyboards/ah/haven60/readme.md:
```diff
@@ -0,0 +1,28 @@
+# haven60
+
+![haven60](https://i.imgur.com/Uja750Oh.png)
```
```suggestion
![haven60](https://i.imgur.com/Uja750Oh.png)

```

CMMS-Freather commented:
@fauxpark @zvecr @waffle87 
If you have time, please take a look and see if there is any other changes I need to make prior to approval

dunk2k commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,118 @@
+            ]
+        }
+    }
+}
```
add End of File
```suggestion
}

```

dunk2k commented on keyboards/ah/haven60/keymaps/via/rules.mk:
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
add EoF
```suggestion
VIA_ENABLE = yes

```

dunk2k commented on keyboards/ah/haven60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,34 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /*
+     * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+     * │ ` │ 1 │ 2 │ 3 │ 4 │ 5 │ 6 │ 7 │ 8 │ 9 │ 0 │ - │ = │ Backsp│
+     * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+     * │ Tab │ Q │ W │ E │ R │ T │ Y │ U │ I │ O │ P │ [ │ ] │  \  │
+     * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤
+     * │ Caps │ A │ S │ D │ F │ G │ H │ J │ K │ L │ ; │ ' │  Enter │
+     * ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────────┤
+     * │ Shift  │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │    Shift │
+     * ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+     * │Ctrl│GUI │Alt │                        │ Alt│   GUI   │Ctrl│
+     * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+     */
```
```suggestion
```

dunk2k commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,34 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /*
+     * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+     * │ ` │ 1 │ 2 │ 3 │ 4 │ 5 │ 6 │ 7 │ 8 │ 9 │ 0 │ - │ = │ Backsp│
+     * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+     * │ Tab │ Q │ W │ E │ R │ T │ Y │ U │ I │ O │ P │ [ │ ] │  \  │
+     * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤
+     * │ Caps │ A │ S │ D │ F │ G │ H │ J │ K │ L │ ; │ ' │  Enter │
+     * ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────────┤
+     * │ Shift  │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │    Shift │
+     * ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+     * │Ctrl│GUI │Alt │                        │ Alt│   GUI   │Ctrl│
+     * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+     */
```
```suggestion
```

dunk2k commented:
To complete compatible Community Layouts, could the matrix co-ordinates (e.g. `0, 2`) for the following keys be given please?:
- 1.75u Right Shift
- 1u key right of 1.75u Right Shift
- left 1u key of Split Backspace
- right 1u key of Split Backspace

CMMS-Freather commented:
> To complete compatible Community Layouts, could the matrix co-ordinates (e.g. `0, 2`) for the following keys be given please?:
> 
> * 1.75u Right Shift
> * 1u key right of 1.75u Right Shift
> * left 1u key of Split Backspace
> * right 1u key of Split Backspace

sure
1.75u is the same as 2.75u right shift
1u key of the split rs is 3,13
left 1u key is same as 2u backspace, 0,13
right 1u key of split bs is 1,13

if you'd like, 
the left shift key is 3,0
1.25u of  the split ls key is 3,0
1u of t he split ls key is 3,1

fauxpark commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,181 @@
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
```
```suggestion
```

dunk2k commented:
> right 1u key of split bs is 1,13

that matrix co-ordinate is identical to the ANSI blackslash key, i.e. a matrix co-ordinate can't be shared with more than one key 🙈
Would that mean the ANSI backslash key has a different matrix co-ordinate when split backspace is used? or right 1u key of split backspace uses a different matrix co-ordinate - maybe starting `4, ?` ?

dunk2k commented:
> if you'd like,
the left shift key is 3,0
1.25u of the split ls key is 3,0
1u of t he split ls key is 3,1

Would this be indicate that there is an ISO layout (ISO Hash and ISO Enter keys) for the solder version of the PCB?

CMMS-Freather commented:
> > right 1u key of split bs is 1,13
> 
> that matrix co-ordinate is identical to the ANSI blackslash key, i.e. a matrix co-ordinate can't be shared with more than one key 🙈 Would that mean the ANSI backslash key has a different matrix co-ordinate when split backspace is used? or right 1u key of split backspace uses a different matrix co-ordinate - looking at the PCB traces with image provided, in `readme.md`, the right split backspace key uses the same column as ANSI backslash, but can't see the row trace.

so in the design, there are a few keys that share the same coordinate
left 1u of split bs with regular 2u bs
1.5u \ key with iso 1u \ key
ansi 2.25u enter with iso enter
1.25u key of split ls and regualr 2.25u ls
left 1.75u key of split rs with regular 2.75u rs
the reason behind is so the matrix is more occupied and no extra pins required to create an extra column

eventually, in via configurator, it wont bother as much

dunk2k commented:
> so in the design, there are a few keys that share the same coordinate
left 1u of split bs with regular 2u bs
1.5u \ key with iso 1u \ key
ansi 2.25u enter with iso enter
1.25u key of split ls and regualr 2.25u ls
left 1.75u key of split rs with regular 2.75u rs

Thankfully the shared keys you've mentioned don't collide.
At current, what's not going to compile due to collision is a layout with 1.5u key (ANSI backslash) and split backspace, as the ANSI backslash and right 1u key of split backspace both use the same matrix co-ordinate (`1, 13`).
In the case of a layout with 1.5u key (ANSI backslash) and split backspace - which key is assigned matrix co-ordinate `1, 13` and which matrix co-ordinate is the remaining key assigned?

dunk2k commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,180 @@
+                {"matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
+                {"matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
+            ]
+        }
```
```suggestion
        },
        "LAYOUT_60_iso_tsangan": {
            "layout": [
                {"matrix": [0, 0], "x": 0, "y": 0},
                {"matrix": [0, 1], "x": 1, "y": 0},
                {"matrix": [0, 2], "x": 2, "y": 0},
                {"matrix": [0, 3], "x": 3, "y": 0},
                {"matrix": [0, 4], "x": 4, "y": 0},
                {"matrix": [0, 5], "x": 5, "y": 0},
                {"matrix": [0, 6], "x": 6, "y": 0},
                {"matrix": [0, 7], "x": 7, "y": 0},
                {"matrix": [0, 8], "x": 8, "y": 0},
                {"matrix": [0, 9], "x": 9, "y": 0},
                {"matrix": [0, 10], "x": 10, "y": 0},
                {"matrix": [0, 11], "x": 11, "y": 0},
                {"matrix": [0, 12], "x": 12, "y": 0},
                {"matrix": [0, 13], "x": 13, "y": 0, "w": 2},
                {"matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"matrix": [1, 1], "x": 1.5, "y": 1},
                {"matrix": [1, 2], "x": 2.5, "y": 1},
                {"matrix": [1, 3], "x": 3.5, "y": 1},
                {"matrix": [1, 4], "x": 4.5, "y": 1},
                {"matrix": [1, 5], "x": 5.5, "y": 1},
                {"matrix": [1, 6], "x": 6.5, "y": 1},
                {"matrix": [1, 7], "x": 7.5, "y": 1},
                {"matrix": [1, 8], "x": 8.5, "y": 1},
                {"matrix": [1, 9], "x": 9.5, "y": 1},
                {"matrix": [1, 10], "x": 10.5, "y": 1},
                {"matrix": [1, 11], "x": 11.5, "y": 1},
                {"matrix": [1, 12], "x": 12.5, "y": 1},
                {"matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"matrix": [2, 1], "x": 1.75, "y": 2},
                {"matrix": [2, 2], "x": 2.75, "y": 2},
                {"matrix": [2, 3], "x": 3.75, "y": 2},
                {"matrix": [2, 4], "x": 4.75, "y": 2},
                {"matrix": [2, 5], "x": 5.75, "y": 2},
                {"matrix": [2, 6], "x": 6.75, "y": 2},
                {"matrix": [2, 7], "x": 7.75, "y": 2},
                {"matrix": [2, 8], "x": 8.75, "y": 2},
                {"matrix": [2, 9], "x": 9.75, "y": 2},
                {"matrix": [2, 10], "x": 10.75, "y": 2},
                {"matrix": [2, 11], "x": 11.75, "y": 2},
                {"matrix": [1, 13], "x": 12.75, "y": 2},
                {"matrix": [2, 13], "x": 13.75, "y": 1, "w": 1.25, "h": 2},
                {"matrix": [3, 0], "x": 0, "y": 3, "w": 1.25},
                {"matrix": [3, 1], "x": 1.25, "y": 3},
                {"matrix": [3, 2], "x": 2.25, "y": 3},
                {"matrix": [3, 3], "x": 3.25, "y": 3},
                {"matrix": [3, 4], "x": 4.25, "y": 3},
                {"matrix": [3, 5], "x": 5.25, "y": 3},
                {"matrix": [3, 6], "x": 6.25, "y": 3},
                {"matrix": [3, 7], "x": 7.25, "y": 3},
                {"matrix": [3, 8], "x": 8.25, "y": 3},
                {"matrix": [3, 9], "x": 9.25, "y": 3},
                {"matrix": [3, 10], "x": 10.25, "y": 3},
                {"matrix": [3, 11], "x": 11.25, "y": 3},
                {"matrix": [3, 12], "x": 12.25, "y": 3, "w": 2.75},
                {"matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
                {"matrix": [4, 1], "x": 1.5, "y": 4},
                {"matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
                {"matrix": [4, 6], "x": 4, "y": 4, "w": 7},
                {"matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"matrix": [4, 12], "x": 12.5, "y": 4},
                {"matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
            ]
        },
        "LAYOUT_60_iso_wkl": {
            "layout": [
                {"matrix": [0, 0], "x": 0, "y": 0},
                {"matrix": [0, 1], "x": 1, "y": 0},
                {"matrix": [0, 2], "x": 2, "y": 0},
                {"matrix": [0, 3], "x": 3, "y": 0},
                {"matrix": [0, 4], "x": 4, "y": 0},
                {"matrix": [0, 5], "x": 5, "y": 0},
                {"matrix": [0, 6], "x": 6, "y": 0},
                {"matrix": [0, 7], "x": 7, "y": 0},
                {"matrix": [0, 8], "x": 8, "y": 0},
                {"matrix": [0, 9], "x": 9, "y": 0},
                {"matrix": [0, 10], "x": 10, "y": 0},
                {"matrix": [0, 11], "x": 11, "y": 0},
                {"matrix": [0, 12], "x": 12, "y": 0},
                {"matrix": [0, 13], "x": 13, "y": 0, "w": 2},
                {"matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"matrix": [1, 1], "x": 1.5, "y": 1},
                {"matrix": [1, 2], "x": 2.5, "y": 1},
                {"matrix": [1, 3], "x": 3.5, "y": 1},
                {"matrix": [1, 4], "x": 4.5, "y": 1},
                {"matrix": [1, 5], "x": 5.5, "y": 1},
                {"matrix": [1, 6], "x": 6.5, "y": 1},
                {"matrix": [1, 7], "x": 7.5, "y": 1},
                {"matrix": [1, 8], "x": 8.5, "y": 1},
                {"matrix": [1, 9], "x": 9.5, "y": 1},
                {"matrix": [1, 10], "x": 10.5, "y": 1},
                {"matrix": [1, 11], "x": 11.5, "y": 1},
                {"matrix": [1, 12], "x": 12.5, "y": 1},
                {"matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"matrix": [2, 1], "x": 1.75, "y": 2},
                {"matrix": [2, 2], "x": 2.75, "y": 2},
                {"matrix": [2, 3], "x": 3.75, "y": 2},
                {"matrix": [2, 4], "x": 4.75, "y": 2},
                {"matrix": [2, 5], "x": 5.75, "y": 2},
                {"matrix": [2, 6], "x": 6.75, "y": 2},
                {"matrix": [2, 7], "x": 7.75, "y": 2},
                {"matrix": [2, 8], "x": 8.75, "y": 2},
                {"matrix": [2, 9], "x": 9.75, "y": 2},
                {"matrix": [2, 10], "x": 10.75, "y": 2},
                {"matrix": [2, 11], "x": 11.75, "y": 2},
                {"matrix": [1, 13], "x": 12.75, "y": 2},
                {"matrix": [2, 13], "x": 13.75, "y": 1, "w": 1.25, "h": 2},
                {"matrix": [3, 0], "x": 0, "y": 3, "w": 1.25},
                {"matrix": [3, 1], "x": 1.25, "y": 3},
                {"matrix": [3, 2], "x": 2.25, "y": 3},
                {"matrix": [3, 3], "x": 3.25, "y": 3},
                {"matrix": [3, 4], "x": 4.25, "y": 3},
                {"matrix": [3, 5], "x": 5.25, "y": 3},
                {"matrix": [3, 6], "x": 6.25, "y": 3},
                {"matrix": [3, 7], "x": 7.25, "y": 3},
                {"matrix": [3, 8], "x": 8.25, "y": 3},
                {"matrix": [3, 9], "x": 9.25, "y": 3},
                {"matrix": [3, 10], "x": 10.25, "y": 3},
                {"matrix": [3, 11], "x": 11.25, "y": 3},
                {"matrix": [3, 12], "x": 12.25, "y": 3, "w": 2.75},
                {"matrix": [4, 0], "x": 0, "y": 4, "w": 1.5},
                {"matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.5},
                {"matrix": [4, 6], "x": 4, "y": 4, "w": 7},
                {"matrix": [4, 11], "x": 11, "y": 4, "w": 1.5},
                {"matrix": [4, 13], "x": 13.5, "y": 4, "w": 1.5}
            ]
        }
```

dunk2k reviewed (changes requested):
add ISO Community Layout support

CMMS-Freather commented:
> > so in the design, there are a few keys that share the same coordinate
> > left 1u of split bs with regular 2u bs
> > 1.5u \ key with iso 1u \ key
> > ansi 2.25u enter with iso enter
> > 1.25u key of split ls and regualr 2.25u ls
> > left 1.75u key of split rs with regular 2.75u rs
> 
> Thankfully the shared keys you've mentioned don't collide. At current, what's not going to compile due to collision is a layout with 1.5u key (ANSI backslash) and split backspace, as the ANSI backslash and right 1u key of split backspace both use the same matrix co-ordinate (`1, 13`).
> 
> ```
> ─┬───┬───┐
>  │0D │1D │
> ─┴─┬─┴───┤
>    │1D   │
> ───┴─────┤
> 
> D=13
> ```
> 
> In the case of a layout with 1.5u key (ANSI backslash) and split backspace - which key is assigned matrix co-ordinate `1, 13` and which matrix co-ordinate is the remaining key assigned?

i think i made a mistake on previous quote reply.
in this specific layout, the left 1u key of the split bs is 0,13, sharing coordinate with 2u backspace
the right 1u key of the split bs is 2,12 ( a bit of weird position right lmao)
then the 1.5u key is still 1,13
this should solve your question
and thanks in advnace of adding community layout

dunk2k commented:
> i think i made a mistake on previous quote reply.
in this specific layout, the left 1u key of the split bs is 0,13, sharing coordinate with 2u backspace
the right 1u key of the split bs is 2,12 ( a bit of weird position right lmao)
then the 1.5u key is still 1,13
this should solve your question
and thanks in advnace of adding community layout

thanks for confirming, more code suggestions inbound 😉

dunk2k commented on keyboards/ah/haven60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_ansi_tsangan(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                     KC_RALT, MO(1),  KC_RCTL
+    ),
+    [1] = LAYOUT_60_ansi_tsangan(
+        KC_GRV,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,    KC_F11, KC_F12,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_UP, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_LEFT, KC_RIGHT,          KC_TRNS,
+        KC_TRNS,          KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_DOWN,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS, KC_TRNS,  KC_TRNS
```
```suggestion
    [0] = LAYOUT_60_all(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, MO(1),   KC_RCTL
    ),
    [1] = LAYOUT_60_all(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,   KC_F12,  _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_UP,    _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_LEFT, KC_RIGHT,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_DOWN,           _______,
        _______, _______, _______,                            _______,                                     _______,  _______, _______
```

dunk2k commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_ansi_tsangan(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, 
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, 
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT, 
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, 
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                     KC_RALT, MO(1),  KC_RCTL 
+    ),
+    [1] = LAYOUT_60_ansi_tsangan(
+        KC_GRV,   KC_F1,      KC_F2,      KC_F3,      KC_F4,      KC_F5,      KC_F6,      KC_F7,      KC_F8,      KC_F9,      KC_F10,     KC_F11, KC_F12,     KC_TRNS,
+        KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_UP,  KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_LEFT,    KC_RIGHT,           KC_TRNS,
+        KC_TRNS,              KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_DOWN,            KC_TRNS,
+        KC_TRNS,  KC_TRNS,    KC_TRNS,                                        KC_TRNS,                                        KC_TRNS,    KC_TRNS,            KC_TRNS
```
```suggestion
    [0] = LAYOUT_60_all(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, MO(1),   KC_RCTL
    ),
    [1] = LAYOUT_60_all(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,   KC_F12,  _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_UP,    _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_LEFT, KC_RIGHT,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_DOWN,           _______,
        _______, _______, _______,                            _______,                                     _______,  _______, _______
```

dunk2k reviewed (changes requested):
- add split bs + split right shift Community Layouts
- add `LAYOUT_all` layout as per PR Checklist
  - amend `default` and `via` keymaps to use `LAYOUT_all`

dunk2k commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, MO(1),   KC_RCTL
+    ),
+    [1] = LAYOUT_60_all(
```
```suggestion
    [0] = LAYOUT_all(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, MO(1),   KC_RCTL
    ),
    [1] = LAYOUT_all(
```

dunk2k reviewed (changes requested):
CI build failure fixes

dunk2k commented on keyboards/ah/haven60/haven60.c:
```diff
@@ -0,0 +1,46 @@
+        rgblight_set_layer_state(1, !led_state.caps_lock);
+    }
+    return res;
+}
```
```suggestion
}

```

dunk2k commented on keyboards/ah/haven60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,21 @@
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL
```
CI build fix
```suggestion
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,
```

dunk2k commented on keyboards/ah/haven60/info.json:
```diff
@@ -0,0 +1,701 @@
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
```
```suggestion
```
